src/SLCore.IntegrationTests/SLCoreTestProcessRunner.cs

```diff
@@ -0,0 +1,132 @@
+
+namespace SonarLint.VisualStudio.SLCore.IntegrationTests;
+
+[ExcludeFromCodeCoverage]
+public sealed class SLCoreTestProcessRunner : IDisposable
+{
+    private readonly string pathToBat;
+    private readonly string rpcLogFilePath;
+    private readonly string stdErrLogFilePath;
+    private readonly bool enableVerboseLogs;
+    private readonly bool enableAutoFlush;
+    private Process process;
+    private readonly CancellationTokenSource cancellationTokenSource = new();
+    private JsonRpcWrapper jsonRpcWrapper;
+    private StreamWriter logFileStream;
+    private StreamWriter errorFileStream;
+    
+    public ISLCoreJsonRpc Rpc { get; private set; }
+
+
+    public SLCoreTestProcessRunner(string pathToBat, 
+        string rpcLogFilePath = null, 
+        string stdErrLogFilePath = null, 
+        bool enableVerboseLogs = false,
+        bool enableAutoFlush = false)
+    {
+        this.pathToBat = pathToBat;
+        this.rpcLogFilePath = rpcLogFilePath;
+        this.stdErrLogFilePath = stdErrLogFilePath;
+        this.enableVerboseLogs = enableVerboseLogs;
+        this.enableAutoFlush = enableAutoFlush;
+    }
+
+    public void Start()
+    {
+        var processStartInfo = new ProcessStartInfo("cmd.exe", $@"/c {pathToBat}")
+        {
+            CreateNoWindow = true,
+            UseShellExecute = false,
+            RedirectStandardInput = true,
+            RedirectStandardOutput = true,
+            RedirectStandardError = true
+        };
+
+        process = new Process { StartInfo = processStartInfo };
+        
+        process.Start();
+        SetUpLogging();
+        
+        jsonRpcWrapper = new JsonRpcWrapper(process.StandardInput.BaseStream, process.StandardOutput.BaseStream);
+        jsonRpcWrapper.TraceSource.Switch.Level = enableVerboseLogs ? SourceLevels.Verbose : SourceLevels.Warning;
+        jsonRpcWrapper.TraceSource.Listeners.Add(logFileStream == null ? new ConsoleTraceListener() : new TextWriterTraceListener(logFileStream));
+
+        Rpc = new SLCoreJsonRpc(jsonRpcWrapper, new RpcMethodNameTransformer());
+    }
+
+    private void SetUpLogging()
+    {
+        if (!string.IsNullOrEmpty(rpcLogFilePath))
+        {
+            logFileStream = new StreamWriter(File.OpenWrite(rpcLogFilePath));
+            logFileStream.AutoFlush = enableAutoFlush;
+        }
+
+        Task.Run(ReadErrorLog).Forget();
+    }
+
+    private void ReadErrorLog()
+    {
+        var token = cancellationTokenSource.Token;
+        var fileLoggingEnabled = stdErrLogFilePath != null;
+        var prefix = fileLoggingEnabled ? string.Empty : "ERR: ";
+        errorFileStream = fileLoggingEnabled ? new StreamWriter(File.OpenWrite(stdErrLogFilePath)){AutoFlush = enableAutoFlush} : null;
+
+
+        while (!token.IsCancellationRequested)
+        {
+            var line = process.StandardError.ReadLine();
+            if (string.IsNullOrEmpty(line)) // potential problem here if it returns too often?
```

Comment:
Since this is test code I don't see as a big problem, we can focus if it ends up to be a problem.